File: apps/android/app/src/main/java/com/ecodeliveryroutes/core/model/NetworkNode.kt
package com.ecodeliveryroutes.core.model

data class HubSummary(
    val id: String,
    val code: String,
    val name: String,
    val city: String?,
    val isActive: Boolean
)

data class DepotSummary(
    val id: String,
    val hubId: String,
    val code: String,
    val name: String,
    val city: String?,
    val isActive: Boolean
)

data class PointSummary(
    val id: String,
    val hubId: String,
    val depotId: String?,
    val code: String,
    val name: String,
    val city: String?,
    val isActive: Boolean
)

File: apps/apple/SharedCore/Sources/SharedCore/Models/NetworkNode.swift
import Foundation

public struct HubSummary: Codable, Identifiable {
    public let id: String
    public let code: String
    public let name: String
    public let city: String?
    public let isActive: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case code
        case name
        case city
        case isActive = "is_active"
    }
}

public struct DepotSummary: Codable, Identifiable {
    public let id: String
    public let hubId: String
    public let code: String
    public let name: String
    public let addressLine: String?
    public let city: String?
    public let isActive: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case hubId = "hub_id"
        case code
        case name
        case addressLine = "address_line"
        case city
        case isActive = "is_active"
    }
}

public struct PointSummary: Codable, Identifiable {
    public let id: String
    public let hubId: String
    public let depotId: String?
    public let code: String
    public let name: String
    public let addressLine: String?
    public let city: String?
    public let isActive: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case hubId = "hub_id"
        case depotId = "depot_id"
        case code
        case name
        case addressLine = "address_line"
        case city
        case isActive = "is_active"
    }
}
